src/pages/Home/HomeContainer.jsx

```diff
@@ -1,49 +1,51 @@
-import React from 'react';
-import Home from './Home';
-import ProtonSDK from '../../utils/proton';
+import React from "react";
+import Home from "./Home";
+import ProtonSDK from "../../utils/proton";
 
 class HomeContainer extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
       windowWidth: 0,
-    }
+    };
   }
 
   componentDidMount() {
     this.updateWindowWidth();
-    window.addEventListener('resize', this.updateWindowWidth);
+    window.addEventListener("resize", this.updateWindowWidth);
     /* istanbul ignore next */
-    document.addEventListener('backToSelector', () => {
+    document.addEventListener("backToSelector", () => {
       this.generateLoginRequest();
     });
   }
 
   updateWindowWidth = () => {
     this.setState({ windowWidth: window.innerWidth });
-  }
+  };
 
   /* istanbul ignore next */
-  generateLoginRequest = async () =>  {
-    const { setLoggedInState } = this.props;
-    try {
-      this.setState({ isLoggingIn: true });
-      const { auth, accountData } = await ProtonSDK.login();
-      if (auth && auth.actor && auth.permission) {
-        setLoggedInState(auth.actor, auth.permission, accountData);
-      }
-      this.setState({ isLoggingIn: false });
-    } catch (e) {
-      console.error(e);
+  generateLoginRequest = async () => {
+    const { setLoggedInState, setErrorState } = this.props;
+    const { auth, accountData, error } = await ProtonSDK.login();
+    if (error || !(auth && auth.actor && auth.permission)) {
+      setErrorState(error || "Login error");
```

Comment:
Let's also update the backup error message to be more user friendly, maybe "Unable to log in. Please try again."